src/js/diff.js

```diff
@@ -939,7 +939,7 @@ fluid.diff.arraysEqual = function (leftArray, rightArray) {
             else if (leftValue === null && rightValue === null) {
                 return true;
             }
-            else if (typeof leftValue === "object") {
+            else if (typeof leftValue === "object" && leftValue !== null) {
```

Comment:
I think these tests now coincide with the behaviour of fluid.isPlainObject(leftValue, true)

Reply:
Done, throughout.